File: csharp/Platform.Incrementers.Tests/IncrementerTests.cs
using Xunit;

namespace Platform.Incrementers.Tests
{
    public class IncrementerTests
    {
        [Fact]
        public void ParameterlessConstructedSetterTest()
        {
            Incrementer<int> incrementer = new Incrementer<int>();
            Assert.Equal(default, incrementer.Result);
        }

        [Fact]
        public void ConstructedWithDefaultValueSetterTest()
        {
            Incrementer<int> incrementer = new Incrementer<int>(9UL);
            Assert.Equal(9UL, incrementer.Result);
        }

        [Fact]
        public void MethodsWithBooleanReturnTypeTest()
        {
            Incrementer<int> incrementer = new Incrementer<int>();
            incrementer.Increment();
            Assert.Equal(1UL, incrementer.Result);
            Assert.True(incrementer.IncrementAndReturnTrue());
            Assert.Equal(2UL, incrementer.Result);
        }

        [Fact]
        public void MethodsWithIntegerReturnTypeTest()
        {
            Incrementer<int, int> incrementer = new Incrementer<int, int>(1);
            incrementer.Increment();
            Assert.Equal(1UL, incrementer.Result);
            Assert.Equal(1, incrementer.IncrementAndReturnTrue());
            Assert.Equal(2UL, incrementer.Result);
        }
    }
}

File: cpp/Platform.Incrementers.Tests/IncrementerTests.cpp
#include <Platform.Incrementers.h>
#include <gtest/gtest.h>

namespace Platform::Incrementers::Tests
{
    TEST(IncrementersTests, ParameterlessConstructedSetterTest)
    {
        auto incrementer = Incrementer<int>();
        ASSERT_EQ({}, incrementer.Result());
    }

    TEST(IncrementersTests, ConstructedWithDefaultValueSetterTest)
    {
        auto incrementer = Incrementer<int>(9UL);
        ASSERT_EQ(9UL, incrementer.Result());
    }

    TEST(IncrementersTests, MethodsWithBooleanReturnTypeTest)
    {
        auto incrementer = Incrementer<int>();
        incrementer.Increment();
        ASSERT_EQ(1UL, incrementer.Result());
        ASSERT_TRUE(incrementer.IncrementAndReturnTrue());
        ASSERT_EQ(2UL, incrementer.Result());
    }

    TEST(IncrementersTests, MethodsWithIntegerReturnTypeTest)
    {
        auto incrementer = Incrementer<int, int>(1);
        incrementer.Increment();
        ASSERT_EQ(1UL, incrementer.Result());
        ASSERT_EQ(1, incrementer.IncrementAndReturnTrue());
        ASSERT_EQ(2UL, incrementer.Result());
    }
}
